cmd/client/main.go

```diff
@@ -290,7 +322,7 @@ func (c *Client) getUDSDialer(o *GrpcProxyClientOptions) (func(ctx context.Conte
 
 	// Setup signal handler
 	ch := make(chan os.Signal, 1)
-	signal.Notify(ch)
+	defer signal.Notify(ch)
```

Comment:
Sorry I don't get why we need to change this to `defer`. Could you explain a little bit?

Reply:
I was getting proxyConn closing early (i.e. during tunnel creation) some of the time and incorrectly attributed it to this and the following goroutine. Good catch.